### DISCUSSION TITLE
Is there a way to re-create the objective function used in Ridge Regression within Robyn - so that I can use it to predict on unseen data?

### DISCUSSION THREAD

**Discussion by shuvayan** [facebookexperimental/Robyn#961] [Q&A]
Hi,

The model refresh part is not working for me currently as it says it cannot find the relevant json file.

[https://github.com/facebookexperimental/Robyn/issues/711](url)

So I was thinking if there is a way I can recreate the objective function using the coefficients , intercept and hyper parameter values as returned by the robyn model.
If yes , I could use it to predict on unseen data.
Kindly let me know if anyone has come up with something similar

**Comment by shantisb** [comment]
Hey @shuvayan were you able to figure this out?

**Comment by AnantChandra** [comment]
There's a workaround by manually recreating the prediction function using the coefficients, intercept, and hyperparameters from your trained model.

Try this:
```
coefficients <- robyn_object$xDecompAgg$decomp.rssd$coef
intercept <- robyn_object$xDecompAgg$decomp.rssd$intercept
lambda <- robyn_object$hyperparameters$lambda

predict_ridge <- function(new_data, coefficients, intercept) {
  predictions <- as.matrix(new_data) %*% coefficients + intercept
  return(predictions)
}

new_predictions <- predict_ridge(new_data, coefficients, intercept)
```

**Reply by shantisb** [reply]
Thank you so much @AnantChandra ! Do you know how can it be done now that robyn_save() (to save as .RDS file ) has been deprecated? Now we only have:

`ExportedModel <- robyn_write(InputCollect, OutputCollect, select_model, export = TRUE)
` which saves the model as a json file not an RDS file. Then you can just read that json file and do  
`json_data <- robyn_read(json_file)` but this json_data doesn't have `xDecompAgg$decomp` that I know of.
